src/components/cart.js

```diff
@@ -96,22 +115,56 @@ export default class Cart extends Component {
       classes: this.classes,
       lineItemsHtml: this.lineItemsHtml,
       isEmpty: this.isEmpty,
-      formattedTotal: this.formattedLineItemsSubtotal,
+      formattedTotal: this.formattedTotal,
+      discounts: this.discounts,
       contents: this.options.contents,
       cartNote: this.cartNote,
     });
   }
 
+  get styles() {
+    return this.options.manifest.filter((component) => this.config[component].styles).reduce((hash, component) => {
+      const styles = this.config[component].styles;
+      if (component === 'cart') {
+        if (styles.subtotal) {
+          styles.discountAmount = styles.subtotal;
+        }
+
+        if (styles.subtotalText) {
+          styles.discountText = styles.subtotalText;
+        }
+      }
+      hash[component] = styles;
+      return hash;
+    }, {});
+  }
+
   /**
    * get formatted cart subtotal based on moneyFormat
    * @return {String}
    */
   get formattedTotal() {
-    return formatMoney(this.model.subtotalPrice, this.moneyFormat);
+    const total = this.options.contents.discounts ? this.model.subtotalPriceV2.amount : this.model.lineItemsSubtotalPrice.amount;
+    return formatMoney(total, this.moneyFormat);
   }
 
-  get formattedLineItemsSubtotal() {
-    return formatMoney(this.model.lineItemsSubtotalPrice.amount, this.moneyFormat);
+  get discounts() {
+    if (!this.options.contents.discounts) {
+      return null;
+    }
+
+    return this.model.discountApplications.reduce((discountArr, discount) => {
+      if (discount.targetSelection === 'ALL') {
+        let discountValue = 0;
+        if (discount.value.amount) {
+          discountValue = discount.value.amount;
+        } else if (discount.value.percentage) {
+          discountValue = (discount.value.percentage / 100) * this.model.lineItemsSubtotalPrice.amount;
+        }
+        discountArr.push({text: discount.title, amount: `- ${formatMoney(discountValue, this.moneyFormat)}`});
```

Comment:
Is there a discount that isn't amount or percentage? If so, should the `discountArr` only update if discount value is greater than zero?